### DISCUSSION TITLE
Preview multiple PRs at once ?

### DISCUSSION THREAD

**Discussion by KaKi87** [vercel/vercel#4929] [Ideas] [answered]
Hello there :)

I've seen the Vercel bot on GitHub in action and the previews feature is really awesome.

I'm currently following the gitmoji project and they are currently 3 specific PRs open for specific website features that I've been very happy to preview.

But, separately ! Currently, I bookmarked one of these previews because I really love the implemented feature, but it would be even better if I could have the three of these in one.

Do you people think this could be ever possible ? Previewing multiple PRs at once ?

Thanks !

**Comment by styfle** [comment, marked as answer]
Yes, you can preview multiple PRs at once since every PR has a unique URL to a preview deployment.

For example, you'll see a preview URL on each PR that looks something like `https://projectname-git-branchname.username.vercel.app`

Make sure you have the GitHub Integration setup so Vercel knows which GitHub user/team should be linked to which Vercel user/team.

https://vercel.com/docs/git/vercel-for-github

**Reply by KaKi87** [reply]
I'm sorry I think you misunderstood me.
What I meant is : for instance, a repository has two PRs, each have a preview URL, but I'd like to have a link that previews both auto-merged.
Thanks

**Reply by styfle** [reply]
Vercel doesn't change your source code or perform merges.

What you can do is create a 3rd PR that represents the 2 PRs in question and perform the merge yourself. Then you'll get a new preview URL with "both" PRs.

**Reply by KaKi87** [reply]
How do I do that ?

**Reply by styfle** [reply]
Use the same git commands you used to create the 1st and 2nd branch to create a 3rd branch.

```sh
git pull
git checkout -b branch3  # create a new branch
git merge branch1  # branch of first PR
git merge branch2 # branch of second PR
git push origin branch3 # create a third PR
```

**Reply by KaKi87** [reply]
As an maintainer or collaborator, sure, but I'm neither :sweat_smile:

**Reply by styfle** [reply]
You would need to fork the GitHub project and then import into Vercel first.

1. Import using this link: https://deploy.new/carloscuesta/gitmoji
2. Follow my git merge steps above by first cloning your fork

**Reply by KaKi87** [reply]
It's done, thanks ! https://gitmoji.kaki87.vercel.app/
However, I couldn't find out how to trigger deploy on this specific [`preview/dark+search`](https://github.com/KaKi87/gitmoji/tree/preview/dark%2Bsearch) branch which I pushed *before* setting up Vercel ?
So I installed the Vercel CLI to manually deploy but it does not recognizes the branch.
Thanks

**Reply by styfle** [reply]
Vercel CLI deploys the code on your machine. So you would need to do `git checkout BRANCHNAME` and then `vercel` to deploy.

Each time you `git push` to a branch, Vercel will deploy.

**Reply by KaKi87** [reply]
> Each time you `git push` to a branch, Vercel will deploy.

Even when `Everything up-to-date` ?

**Reply by styfle** [reply]
No, that means the push didn't succeed because nothing changed. Try changing a file and committing and then push.

**Reply by KaKi87** [reply]
No, as I said, I pushed the changes BEFORE setting up Vercel.
So I'm trying to deploy the already existing but not deployed branch at its current state.
Thanks
